---
title: "GSE74548"
author: "hcy_meth meta-analysis group"
date: "21/07/2020"
output:
  html_document:
    toc: true
theme: cosmo
---

##  Introduction

In this report, I will take you through a re-analysis methylation data first described by 
[Kok et al (2015)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4644301/).

In their study, they analysed the DNA methylation patterns of 87 participants of aged 65-75 years with midly elevated Homocysteine levels. THese individuals were randomly asigned to take 400 μg folic acid and 500 μg vitamin B12 per day or a placebo during an intervention period of 2 years.

The platform used in the study is the Illumina Infinium HumanMethylation450k BeadChip assay.
The authors used a pipeline based on DMRs 
([Peters et al, 2017](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4429355/)), together with Benjamini-Hochberg(BH) procedure ([Benjamini et al, 1995](https://rss.onlinelibrary.wiley.com/doi/abs/10.1111/j.2517-6161.1995.tb02031.x)).

The methylation data have been deposited to NCBI GEO repository accession number 
[GSE74548](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74548)

The main conclusions from the original study were:

* Long-term supplementation with folic acid and vitamin B12 resulted in DNA methylation changes in leukocytes of older persons. 

* A change in DNA methylation was observed to be different between the participants receiving folic acid and vitamin B12 versus placebo.

* DNA methylation levels of several genomic loci were found to correlate to serum levels of either folate, vitamin B12, or plasma homocysteine. 

* Most prominent DNA methylation patterns associated with supplemental intake or status of B-vitamins are related to developmental processes as well as carcinogenesis.

The aim of this work is to;

1. Asses the relationship between Homocysteine to DNA methylation patterns

2. develop the analytical pipelines required for efficient re-analysis of 450K array data,

3. to confirm that we are able to obtain differential methylation results that are similar
to those obtained in the original study, and

4. to critically evaluate the conclusions made in the original study. 

## Loading packages

These packackes will help us to perform vital steps such as normalisation, filtering, 
differential analysis, etc, and provide information about the array probe annotaions.

```{r,packages}
suppressPackageStartupMessages({
    library("plyr")
    library("R.utils")
    library("missMethyl")
    library("limma")
    library("topconfects")
    library("minfi")
    library("IlluminaHumanMethylation450kmanifest")
    library("RColorBrewer")
    library("IlluminaHumanMethylation450kanno.ilmn12.hg19")
    library("eulerr")
    library("plyr")
    library("gplots")
    library("reshape2")
    library("beeswarm")
  })
# Annotation
ann450k <- getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
myann <- data.frame(ann450k[,c("UCSC_RefGene_Name","Regulatory_Feature_Group")])
promoters <- grep("Prom",myann$Regulatory_Feature_Group)
```
## Loading functions

These functions provide shortcuts to help with charts and other analysis. They will
eventually be shoved into another Rscript or package but can stay here for now.

```{r,load_functions}
source("https://raw.githubusercontent.com/markziemann/ART_methylation/master/meth_functions.R")
```
## Data import

Data will be imported from GEO data base under the accession number-GSE74548
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74548
```{r,data}

dir.create("GSE74548")

ARRAY_SAMPLESHEET="GSE74548/GSE74548_Sample_description_BProof.txt"
# only download it if it is not present on the system
if ( !file.exists(ARRAY_SAMPLESHEET ) ) {
    DLFILE=paste(ARRAY_SAMPLESHEET,".gz",sep="")
    download.file("https://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74548/suppl/GSE74548_Sample_description_BProof.txt.gz",
        destfile = DLFILE)
    gunzip(DLFILE)
}

ARRAY_DATA="GSE74548/GSE74548_RAW.tar"
# only download it if it is not present on the system
if ( !dir.exists("GSE74548/IDAT") ) {
  dir.create("GSE74548/IDAT")
  download.file("https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE74548&format=file",
    destfile = ARRAY_DATA)
    untar(exdir = "GSE74548/IDAT", tarfile = ARRAY_DATA)
}

baseDir <- "GSE74548"
R.utils::gunzip("GSE74548/IDAT/GPL13534_HumanMethylation450_15017482_v.1.1.csv.gz",overwrite=TRUE, remove=FALSE)
targets <- read.metharray.sheet(baseDir,pattern="csv")
download.file("https://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74548/matrix/GSE74548_series_matrix.txt.gz",destfile = "GSE74548_series_matrix.txt.gz")

gse<- getGEO(filename = "GSE74548_series_matrix.txt.gz")

targets <- pData(phenoData(gse))
targets <- targets[order(rownames(targets)),]
mybase <- unique(gsub("_Red.idat.gz" ,"", gsub("_Grn.idat.gz", "" ,list.files("./GSE74548",pattern = "GSM",recursive = TRUE))))
mybase <- paste("GSE74548/", mybase, sep = "")
# sample number discrepancy. 311 IDAT FILES but GEO states 174 subjects.
gsm <-sapply(strsplit(mybase,"_"),"[[",1)
gsm <-gsub("GSE74548/IDAT/","",gsm)
#only using sampels described in meta data
targets$Basename<- mybase[which(gsm %in% rownames(targets))]
rgSet <- read.metharray.exp(targets = targets)
rgSet
head(targets)

```
## Normalisation

```{r,norm,fig.cap="Figure 1. Normalisation of bead-array data with SWAN."}
mSet <- preprocessRaw(rgSet)
mSetSw <- SWAN(mSet,verbose=FALSE)
par(mfrow=c(1,2), cex=0.8)
densityByProbeType(mSet[,1], main = "Raw")
densityByProbeType(mSetSw[,1], main = "SWAN")
```
## Filter probes

Here we are running parallel analyses, both including and excluding sex chromosomes.

```{r,filterprobes}
# include sex chromosomes
detP <- detectionP(rgSet)
keep <- rowSums(detP < 0.01) == ncol(rgSet)
mSetSw <- mSetSw[keep,]

# exclude SNP probes
mSetSw <- mapToGenome(mSetSw)
mSetSw_nosnp <- dropLociWithSnps(mSetSw)
dim(mSetSw)
dim(mSetSw_nosnp)
mSetSw <- mSetSw_nosnp

# exclude sex chromosomes
keep <- !(featureNames(mSetSw) %in% ann450k$Name[ann450k$chr %in% c("chrX","chrY")])
mSetFlt <- mSetSw[keep,]
head(mSetFlt)
dim(mSetFlt)
```
## Extracting Beta and M-values

```{r,beta_m_vals}
# include sex chromosomes
meth <- getMeth(mSetSw)
unmeth <- getUnmeth(mSetSw)
Mval <- log2((meth + 100)/(unmeth + 100))
beta <- getBeta(mSetSw)

# exclude sex chromosomes
meth <- getMeth(mSetFlt)
unmeth <- getUnmeth(mSetFlt)
Mval_flt <- log2((meth + 100)/(unmeth + 100))
beta_flt <- getBeta(mSetFlt)
```

## MDS analysis
[Multidimensional scaling(https://en.wikipedia.org/wiki/Multidimensional_scaling)  plot is a method used
to identify the major sources of variation in a dataset. In the MDS plots below, I will be plotting the 
first two dimensions (principal components [PCs]), with each sample label coloured either by Hcy 
classification, sample group,age, folate levels and vitb12 levels.

We will begin with MDS analysis including the sex chromosomes and then exclude them.

First, let's quantify the contribution of the major principal components. with a scree plot, so we can see 
whether most of the variation is captured in the first two PCs or whether it is spread over more PCs.
As we can see in Figure 2, the main source of variation is what is shown in PC1, and a much lesser
extent on the other dimensions. Interestingly, excluding sex chromosomes does not seem to change the 
relative contributions of PCs very much.

```{r,scree,fig.cap="Figure 2. Screeplot shows contribution of top principal components to the overall variation in the experiment."}
colnames(Mval)<-sapply(strsplit(colnames(Mval),"_"),"[[",1)
colnames(Mval_flt)<-sapply(strsplit(colnames(Mval_flt),"_"),"[[",1)
head(Mval)
head(Mval_flt)
par(mfrow=c(2,1))  
myscree(Mval,main="incl sex chr")
myscree(Mval_flt,main="excl sex chr")
```
```{r,defected rows which is not identified as either male or female}

GSM1922496<-which(targets$geo_accession== "GSM1922496")
GSM1922576<-which(targets$geo_accession== "GSM1922576")
GSM1922578<-which(targets$geo_accession== "GSM1922578")
GSM1922618<-which(targets$geo_accession== "GSM1922618")
GSM1922648<-which(targets$geo_accession== "GSM1922648")
defects<-c(GSM1922496,GSM1922576,GSM1922578,GSM1922618,GSM1922648)
defects<-data.frame(defects)
defects

which(colnames(targets)== "characteristics_ch1.4")

#detecting the sex of the defected rows
GSM1922496<-targets[3,14]
GSM1922576<-targets[83,14]
GSM1922578<-targets[85,14]
GSM1922618<-targets[125,14]
GSM1922648<-targets[155,14]

defects <- c(GSM1922496,GSM1922576,GSM1922578,GSM1922618,GSM1922648)
defects<-data.frame(defects)
row.names(defects)<-c("GSM1922496",
                      "GSM1922576",
                      "GSM1922578",
                      "GSM1922618",
                      "GSM1922648")
library(knitr)
kable(defects)
defects
which(row.names(targets)=="GSM1922638")
targets[83,14]
targets[3,14]
targets[5,14]
targets[105,14]
targets[125,14]
```

```{r,updated target after eliminating defects}                                       
targets<-targets[-c(3,83,85,125,155), ]
updated<-targets[-c(83,3,5,105,125),]
str(targets)
dim(updated)
```



```{r,mds1,fig.width = 8 ,fig.height = 8,fig.cap="Figure 3. MDS plot coloured by targets groups"}
head(targets)
dim(targets)
targets$group <- factor(targets$source_name_ch1)
sample_group<-factor(targets$group)
targets$sex <- factor(targets$`gender:ch1`)
dim(sample_group)

colour_palette=brewer.pal(n = length(levels(targets$group)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$source_name_ch1))]
plot(1,axes = FALSE,xlab="",ylab="",main="treatment groups")
legend("center",legend=levels(targets$group),pch=16,cex=1.2,col=colour_palette)
mydist <- plotMDS(Mval, labels=targets$geo_accession,col=colours,main="sex chromosomes included")
mydist_flt <- plotMDS(Mval_flt, labels=targets$geo_accession,col=colours,main="sex chromosomes excluded")
```

```{r,hist1}
hcy<- targets$characteristics_ch1.10
hcy
hcy<-sapply(strsplit(hcy," "),"[[",5)
hcy<-as.numeric(hcy)
hist(hcy,breaks = 20,xlab = "hcy levels")
hcy_groups<-cut(hcy,breaks = c(0,10,16,30),labels = c("low","medium","high"))
table(hcy_groups)
```

```{r,mds2,fig.width = 8 ,fig.height = 8,fig.cap="Figure 3. MDS plot coloured by homocysteine levels"}
head(targets)
targets$sex <- factor(targets$`gender:ch1`)
targets$hcy_groups <- hcy_groups
sample_group<-hcy_groups

colour_palette=brewer.pal(n = length(levels(sample_group)), name = "Paired")
colours <- colour_palette[as.integer(factor(hcy_groups))]
plot(1,axes = FALSE,xlab="",ylab="",main="hcy levels")
legend("center",legend=levels(hcy_groups),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$Sample_Name,col=colours,main="sex chromosomes included")
plotMDS(mydist_flt, labels=targets$Sample_Name,col=colours,main="sex chromosomes excluded")
```

```{r,hist2}
age<- targets$characteristics_ch1.5
age
age<-sapply(strsplit(age," "),"[[",4)
age<-as.numeric(age)
hist(age,breaks = 20,xlab = "hcy levels")
age_groups<-cut(age,breaks = c(0,65,70,75),labels = c("old","older","oldest"))
table(age_groups)
```

```{r,mds3,fig.width = 8 ,fig.height = 8,fig.cap="Figure 3. MDS plot coloured by age"}
head(targets)
targets$age <- factor(age_groups)
sample_group <-factor(age_groups)
colour_palette=brewer.pal(n = length(levels(sample_group)), name = "Paired")
colours <- colour_palette[as.integer(factor(age_groups))]
plot(1,axes = FALSE,xlab="",ylab="",main="age")
legend("center",legend=levels(age_groups),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$Sample_Name,col=colours,main="sex chromosomes included")
plotMDS(mydist_flt,labels=targets$Sample_Name,col=colours,main="sex chromosomes excluded")
```

```{r,hist3}
folate_levels<- targets$characteristics_ch1.8
folate_levels
folate_levels<-sapply(strsplit(folate_levels," "),"[[",5)
folate_levels<-as.numeric(folate_levels)
hist(folate_levels,breaks = 20,xlab = "folate levels")
folate_levels<-cut(folate_levels,breaks = c(0,20,35,50,70,90),labels = c("lowest","low","medium","high","highest"))
table(folate_levels)
folate_levels
```

```{r,mds4,fig.width = 8 ,fig.height = 8,fig.cap="Figure 3. MDS plot coloured by folate levels"}
targets$folate_levels <- folate_levels
sample_group<-folate_levels

colour_palette=brewer.pal(n = length(levels(sample_group)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$folate_levels))]
plot(1,axes = FALSE,xlab="",ylab="",main="Folate levels")
legend("center",legend=levels(targets$folate_levels),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$Sample_Name,col=colours,main="sex chromosomes included")
plotMDS(mydist_flt, labels=targets$Sample_Name,col=colours,main="sex chromosomes excluded")
```

```{r,hist4}
vitb12_levels<- targets$characteristics_ch1.9
vitb12_levels
vitb12_levels<-sapply(strsplit(vitb12_levels," "),"[[",6)
vitb12_levels<-as.numeric(vitb12_levels)
hist(vitb12_levels,breaks = 20,xlab = "vitb12 levels")
vitb12_levels<-cut(vitb12_levels,breaks = c(100,300,500,700,900,1115),labels = c("lowest","low","medium","high","highest"))
table(vitb12_levels)
vitb12_levels
```

```{r,mds5,fig.width = 8 ,fig.height = 8,fig.cap="Figure 3. MDS plot coloured by vitaminb12 levels"}
targets$vitb12_levels <- vitb12_levels
sample_group<-vitb12_levels

colour_palette=brewer.pal(n = length(levels(sample_group)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$vitb12_levels))]
plot(1,axes = FALSE,xlab="",ylab="",main="vitb12 levels")
legend("center",legend=levels(targets$vitb12_levels),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$Sample_Name,col=colours,main="sex chromosomes included")
plotMDS(mydist_flt, labels=targets$Sample_Name,col=colours,main="sex chromosomes excluded")
```






# Differential analysis
There are several differential contrasts that would be of interest to us in this study:

* placebo(43) vs supplement(44) follow-up

* placebo(43) vs supplement(44) Baseline

* hcy levels(172)

* folate levels(172)

* vitb12 levels(172)


```{r,dm1A}
samplesheet<-targets[grep("follow-up",targets$source_name_ch1),]
sex <- factor(samplesheet$`gender:ch1`)
age<-sapply(strsplit(samplesheet$characteristics_ch1.5," "),"[[",4)
age<-as.numeric(age)
age
groups<-factor(samplesheet$source_name_ch1,levels = c("Buffy coat, placebo, baseline","Buffy coat, FA/vB12, baseline"))
mx <-Mval_flt
name="placebo vs supplement follow-up"
head(myann)

                                                                                      
head(myann)

design <- model.matrix(~ sex + groups)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=3, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma1a <- dma[order(dma$P.Value),]
    dma1a
    head(dm)
    head(dma1a)
dm1a_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
dm1a_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))
```
    
```{r,dm1B}
design <- model.matrix(~ age+ sex +groups)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=4, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma1b <- dma[order(dma$P.Value),]
    head(dm)
    head(dma1b)
    
dm1b_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
dm1b_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))
```


```{r,dm2A}
samplesheet<-targets[grep("baseline",targets$source_name_ch1),]
sex <- factor(samplesheet$`gender:ch1`)
groups<-factor(samplesheet$source_name_ch1,levels = c("Buffy coat, placebo, baseline","Buffy coat, FA/vB12, baseline"))
mx <-Mval_flt
name="placebo vs supplement Baseline"
head(myann)


design <- model.matrix(~ sex + groups)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=3, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma2a <- dma[order(dma$P.Value),]
    head(dma2a)
dm2a_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
dm2a_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))
```

```{r,dm2B}
design <- model.matrix(~ age+ sex +groups)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=4, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma2b <- dma[order(dma$P.Value),]
    head(dm)
    head(dma2b)
    
dm1b_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
dm1b_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))
```


```{r,dm3A}
samplesheet<-targets
hcy<- samplesheet$characteristics_ch1.10
hcy<-sapply(strsplit(hcy," "),"[[",5)
hcy<- as.numeric(hcy)
head(samplesheet)
samplesheet<-samplesheet[which(!is.na(hcy)),]
hcy<-hcy[which(!is.na(hcy))]
hcy

sex <- factor(samplesheet$`gender:ch1`)
head(samplesheet)

age<-sapply(strsplit(samplesheet$characteristics_ch1.5," "),"[[",4)
age<-as.numeric(age)
age
mx <-Mval_flt
name="hcy levels "
head(myann)
head(mx)

design <- model.matrix(~ age+ sex + hcy)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=4, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma3a <- dma[order(dma$P.Value),]
    head(dm)
    head(dma3a)
 dm3a_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
 dm3a_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))   
```


```{r,dm3B}
design <- model.matrix(~ sex + hcy)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=3, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma3b <- dma[order(dma$P.Value),]
    head(dma3b)
 dm3b_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
 dm3b_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))   
    ```

```{r,dm3BB}
design <- model.matrix(~ age + hcy)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=3, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma3bb <- dma[order(dma$P.Value),]
    head(dma3bb)
 dm3bb_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
 dm3bb_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))   
    ```

```{r,dm4a}
samplesheet<-targets
folate_levels<- samplesheet$characteristics_ch1.8
folate_levels<-sapply(strsplit(folate_levels," "),"[[",5)
folate_levels<- as.numeric(folate_levels)
head(samplesheet)

sex <- factor(samplesheet$`gender:ch1`)
head(samplesheet)

age<-sapply(strsplit(samplesheet$characteristics_ch1.5," "),"[[",4)
age<-as.numeric(age)
age

mx <-Mval_flt
name="folate levels"
head(myann)

design <- model.matrix(~ sex + folate_levels)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=3, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma4a <- dma[order(dma$P.Value),]
    head(dma4a)
dm4a_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
dm4a_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))
```

```{r,dm4B}    
design <- model.matrix(~ age+ sex + folate_levels)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=4, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma4b <- dma[order(dma$P.Value),]
    head(dm)
    head(dma4b)
    dm4b_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
    dm4b_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))
```

```{r,dm5a}
samplesheet<-targets
vitminb12<- samplesheet$characteristics_ch1.9
vitminb12<-sapply(strsplit(vitminb12," "),"[[",6)
vitminb12<- as.numeric(vitminb12)
head(samplesheet)

sex <- factor(samplesheet$`gender:ch1`)
head(samplesheet)

age<-sapply(strsplit(samplesheet$characteristics_ch1.5," "),"[[",4)
age<-as.numeric(age)
age

mx <-Mval_flt
name="vitmin b12 levels"
head(myann)

design <- model.matrix(~ sex + vitminb12)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=3, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma5a <- dma[order(dma$P.Value),]
    head(dma5a)
    dm5a_d<-nrow(subset(dm,adj.P.Val<0.05,logFC<0))
    dm5a_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))
    
    ```
    
```{r,dm5b}
design <- model.matrix(~ age+ sex + vitminb12)
    mxs <- mx[,which( colnames(mx) %in% rownames(samplesheet) )]
    fit.reduced <- lmFit(mxs,design)
    fit.reduced <- eBayes(fit.reduced)
    summary(decideTests(fit.reduced))
    dm <- topTable(fit.reduced,coef=4, number = Inf)
    dma <- merge(myann,dm,by=0)
    dma5b <- dma[order(dma$P.Value),]
    head(dm)
    head(dma5b)
    dm5b_d<- nrow(subset(dm,adj.P.Val<0.05,logFC<0))
    dm5b_u<-nrow(subset(dm,adj.P.Val<0.05,logFC>0))
    
```

```{r,summarised dms}                                                        
downs<-c(dm1a_d,dm1b_d,dm2b_d,dm2a_d, dm3a_d,dm3b_d,dm3bb_d, dm4a_d,dm4b_d,dm5a_d,dm5b_d)
ups<-c(dm1a_u,dm1b_u,dm2b_u,dm2a_u, dm3a_u,dm3b_u,dm3bb_u, dm4a_u,dm4b_u,dm5a_u,dm5b_u)
my_summary<-data.frame(downs,ups)
row.names(my_summary)<-c("placebo vs supplement follow-up(sex)",1a
                         "placebo vs supplement follow-up(sex,age)",1b
                         "placebo vs supplement Baseline(sex)",2a
                         "placebo vs supplement Baseline(sex,age)",2b
                         "hcy levels(sex)",3b
                         "hcy levels(sex,age)",3a
                         "hcy levels(age)",3bb
                         "folate levels(sex)",4a
                         "folate levels(sex,age)",4b
                         "vitb12 levels(sex)",5a
                         "vitb12 levels(sex,age)")5b
library(knitr)
kable(my_summary)
```
## Venn diagrams of the differential methylated probes for each contrast

First, we look at the similarity of dmps altered by fresh and frozen procedures. The overlap is very large. 
This indicates a high degree of similarity between the profiles.

```{r,venn1,fig.cap="Comparison of probes in related to placebo vs supplement in follow-up(sex)", fig.width = 8 ,fig.height = 8}
v1 <- list("plb v sup fl(up)" =dm1a_u , 
           "hcy(up)" =dm3a_d ,
           "plb v sup fl(dwn)" =dm1a_d ,
           "hcy(dwn)" =dm3b_d)
head(v1)
plot(euler(v1, shape = "ellipse"), quantities = TRUE)
```

```{r,venn2,fig.cap="Comparison of probes in related to placebo vs supplement in follow-up(sex,age)", fig.width = 8 ,fig.height = 8}
v2 <- list("plb v sup fl(up)" =dm1b_u , 
           "hcy(up)" =dm3a_d ,
           "plb v sup fl(dwn)" =dm1b_d ,
           "hcy(dwn)" =dm3b_d)
head(v2)
plot(euler(v2, shape = "ellipse"), quantities = TRUE)
```


```{r,venn3,fig.cap="Comparison of probes in related to hcy(sex) vs folate level(sex,age)", fig.width = 8 ,fig.height = 8}
v3 <- list("fol(up)" =dm4b_u, 
           "hcy(up)" =dm3a_u,
           "fol(dwn)" =dm4b_d,
           "hcy(dwn)" =dm3a_d)
head(v3)
str(v3)
plot(euler(v3, shape = "ellipse"), quantities = TRUE)

```



```{r,venn4,fig.cap="Comparison of probes in related to hcy(age) vs folate level(sex,age)", fig.width = 8 ,fig.height = 8}
v4 <- list("plb v sup fl(up)" =dm1b_u, 
           "fol(up)" = dm4b_u ,
           "plb v sup fl(dwn)" =dm1b_d ,
           "fol(dwn)" =dm4b_d)
head(v4)
plot(euler(v4, shape = "ellipse"), quantities = TRUE)
```

```{r,venn5,fig.cap="Comparison of probes in related to hcy vs folate level(sex,age)", fig.width = 8 ,fig.height = 8}
v5 <- list("hcy(up)" =dm3a_u, 
           "fol(up)" = dm4b_u ,
           "hcy(dwn)" =dm3a_d ,
           "fol(dwn)" =dm4b_d)
head(v5)
plot(euler(v5, shape = "ellipse"), quantities = TRUE)
```
```{r,venn6,fig.cap="Comparison of probes in related to hcy vs vitb12 level(sex,age)", fig.width = 8 ,fig.height = 8}
v6<- list("hcy(up)" =dm3a_u, 
           "vitb12(up)" = dm5b_u ,
           "hcy(dwn)" =dm3a_d ,
           "vitb12(dwn)" =dm5b_d)
head(v6)
plot(euler(v6, shape = "ellipse"), quantities = TRUE)
```
## Spearman correlations of each contrast
While the above Venn diagrams are suggestive of similarity between contrasts, the best 
way to assess this is with correlation analysis. I have chosen Spearman as based on
the directional p-value ranking metrics

```{r,spearman, fig.width = 8 ,fig.height = 8}
mycontrasts <- list("placebo vs supplement follow-up(sex,age)"=dma1b,"placebo vs supplement Baseline(sex,age)"=dma2b,"hcy levels(sex,age)"=dma3a,"folate levels(sex,age)"=dma4b,"vitb12 levels(sex,age)"=dma5b)

myrnks <- lapply(X = mycontrasts, FUN = myranks)
df <- join_all(myrnks,by="rn")
rownames(df) <- df$rn
df$rn=NULL
colnames(df) <- names(mycontrasts)
head(df)
mycors <- cor(df,method = "spearman")
my_palette <- colorRampPalette(c("darkred","red", "orange", "yellow","white"))(n = 25)
heatmap.2(mycors,scale="none",margin=c(10, 10),cexRow=0.8,trace="none",cexCol=0.8,
    col=my_palette,main="Spearman correlations")
mycors
```


"placebo vs supplement follow-up(sex)",1a
                         "placebo vs supplement follow-up(sex,age)",1b
                         "placebo vs supplement Baseline(sex)",2a
                         "placebo vs supplement Baseline(sex,age)",2b
                         "hcy levels(sex)",3b
                         "hcy levels(sex,age)",3a
                         "hcy levels(age)",3bb
                         "folate levels(sex)",4a
                         "folate levels(sex,age)",4b
                         "vitb12 levels(sex)",5a
                         "vitb12 levels(sex,age)")5b
library(knitr)




































